chainerx_cc/examples/mnist_cc/mnist.cc

```diff
@@ -0,0 +1,85 @@
+#include "mnist.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cstdint>
+#include <fstream>
+#include <memory>
+#include <string>
+
+#include "chainerx/array.h"
+#include "chainerx/device.h"
+#include "chainerx/dtype.h"
+#include "chainerx/routines/creation.h"
+#include "chainerx/shape.h"
+
+namespace {
+
+template <typename T>
+T ReadValue(std::ifstream& ifs) {
```

Comment:
From the function name, one would expect that it takes the byte order into account.
Also, generic implementation is not required in this example.
I suggest only defining `int32_t` and `uint32_t` specialization.
(`int32_t` one would depend on `uint32_t`, as I suggested in the previous review.)